Microsoft.Azure.CosmosRepository/src/Paging/IPage.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.CosmosRepository.Paging
+{
+    /// <summary>
+    /// Represents a page of data from a cosmos query
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public interface IPage<out T> where T : IItem
+    {
+        /// <summary>
+        /// The total amount items that matched the query.
+        /// </summary>
+        int Total { get; }
+
+        /// <summary>
+        /// The page number.
+        /// </summary>
+        int Number { get; }
+
+        /// <summary>
+        /// The size of the current page.
+        /// </summary>
+        int Size { get; }
+
+        /// <summary>
+        /// The items that are in the current page.
+        /// </summary>
+        IEnumerable<T> Items { get; }
```

Comment:
```suggestion
        IReadOnlyList<T> Items { get; }
```

Since we are materializing a list, perhaps we should return an `IReadOnlyList`?

Reply:
Yes I think that makes sense